{
  "uid" : "c35103994b0c3821",
  "name" : "Verify redirection to specialists page not authorized user.",
  "fullName" : "tests.main_page_test.TestMainPage#test_mp_05_verify_redirection_to_specialists_page",
  "historyId" : "3c31b28278473c09e9802cf34a761fb2",
  "time" : {
    "start" : 1728610985348,
    "stop" : 1728610985348,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=129.0.6668.89)\nStacktrace:\n#0 0x55631198002a <unknown>\n#1 0x5563116665e0 <unknown>\n#2 0x55631165e7eb <unknown>\n#3 0x55631164ea09 <unknown>\n#4 0x556311650744 <unknown>\n#5 0x55631164ecbd <unknown>\n#6 0x55631164e52f <unknown>\n#7 0x55631164e422 <unknown>\n#8 0x55631164c493 <unknown>\n#9 0x55631164cada <unknown>\n#10 0x556311668f41 <unknown>\n#11 0x5563116fa8c5 <unknown>\n#12 0x5563116dab22 <unknown>\n#13 0x5563116f9d7d <unknown>\n#14 0x5563116da8c3 <unknown>\n#15 0x5563116a86b3 <unknown>\n#16 0x5563116a968e <unknown>\n#17 0x55631194aa2b <unknown>\n#18 0x55631194e9b1 <unknown>\n#19 0x556311937225 <unknown>\n#20 0x55631194f532 <unknown>\n#21 0x55631191c38f <unknown>\n#22 0x55631196ef28 <unknown>\n#23 0x55631196f0f3 <unknown>\n#24 0x55631197ee7c <unknown>\n#25 0x7f0e56894ac3 <unknown>",
  "statusTrace" : "driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"2fadf8c0f6d698aa078854f54815132a\")>\n\n    @pytest.fixture()\n    @allure.step(f'Open page: {MainPageLinks.URL_MAIN_PAGE}')\n    def main_page_open(driver):\n>       driver.get(MainPageLinks.URL_MAIN_PAGE)\n\ntests/conftest.py:22: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:353: in get\n    self.execute(Command.GET, {\"url\": url})\n/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:344: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f3bf8a56720>\nresponse = {'status': 500, 'value': '{\"value\":{\"error\":\"unknown error\",\"message\":\"unknown error: net::ERR_CONNECTION_REFUSED\\\\n  ...\\\\n#23 0x55631196f0f3 \\\\u003Cunknown>\\\\n#24 0x55631197ee7c \\\\u003Cunknown>\\\\n#25 0x7f0e56894ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE         (Session info: chrome=129.0.6668.89)\nE       Stacktrace:\nE       #0 0x55631198002a <unknown>\nE       #1 0x5563116665e0 <unknown>\nE       #2 0x55631165e7eb <unknown>\nE       #3 0x55631164ea09 <unknown>\nE       #4 0x556311650744 <unknown>\nE       #5 0x55631164ecbd <unknown>\nE       #6 0x55631164e52f <unknown>\nE       #7 0x55631164e422 <unknown>\nE       #8 0x55631164c493 <unknown>\nE       #9 0x55631164cada <unknown>\nE       #10 0x556311668f41 <unknown>\nE       #11 0x5563116fa8c5 <unknown>\nE       #12 0x5563116dab22 <unknown>\nE       #13 0x5563116f9d7d <unknown>\nE       #14 0x5563116da8c3 <unknown>\nE       #15 0x5563116a86b3 <unknown>\nE       #16 0x5563116a968e <unknown>\nE       #17 0x55631194aa2b <unknown>\nE       #18 0x55631194e9b1 <unknown>\nE       #19 0x556311937225 <unknown>\nE       #20 0x55631194f532 <unknown>\nE       #21 0x55631191c38f <unknown>\nE       #22 0x55631196ef28 <unknown>\nE       #23 0x55631196f0f3 <unknown>\nE       #24 0x55631197ee7c <unknown>\nE       #25 0x7f0e56894ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:229: WebDriverException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "driver",
    "time" : {
      "start" : 1728610985348,
      "stop" : 1728610986123,
      "duration" : 775
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "clear_allure_results_folder",
    "time" : {
      "start" : 1728610900918,
      "stop" : 1728610900918,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "main_page_open",
    "time" : {
      "start" : 1728610986123,
      "stop" : 1728610986545,
      "duration" : 422
    },
    "status" : "broken",
    "statusMessage" : "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=129.0.6668.89)\nStacktrace:\n#0 0x55631198002a <unknown>\n#1 0x5563116665e0 <unknown>\n#2 0x55631165e7eb <unknown>\n#3 0x55631164ea09 <unknown>\n#4 0x556311650744 <unknown>\n#5 0x55631164ecbd <unknown>\n#6 0x55631164e52f <unknown>\n#7 0x55631164e422 <unknown>\n#8 0x55631164c493 <unknown>\n#9 0x55631164cada <unknown>\n#10 0x556311668f41 <unknown>\n#11 0x5563116fa8c5 <unknown>\n#12 0x5563116dab22 <unknown>\n#13 0x5563116f9d7d <unknown>\n#14 0x5563116da8c3 <unknown>\n#15 0x5563116a86b3 <unknown>\n#16 0x5563116a968e <unknown>\n#17 0x55631194aa2b <unknown>\n#18 0x55631194e9b1 <unknown>\n#19 0x556311937225 <unknown>\n#20 0x55631194f532 <unknown>\n#21 0x55631191c38f <unknown>\n#22 0x55631196ef28 <unknown>\n#23 0x55631196f0f3 <unknown>\n#24 0x55631197ee7c <unknown>\n#25 0x7f0e56894ac3 <unknown>\n\n",
    "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/_pytest/fixtures.py\", line 1123, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/_pytest/fixtures.py\", line 902, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n                     ^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/auto-tests-python/auto-tests-python/tests/conftest.py\", line 22, in main_page_open\n    driver.get(MainPageLinks.URL_MAIN_PAGE)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 353, in get\n    self.execute(Command.GET, {\"url\": url})\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 344, in execute\n    self.error_handler.check_response(response)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n",
    "steps" : [ {
      "name" : "Open page: https://brainup.site/",
      "time" : {
        "start" : 1728610986123,
        "stop" : 1728610986544,
        "duration" : 421
      },
      "status" : "broken",
      "statusMessage" : "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=129.0.6668.89)\nStacktrace:\n#0 0x55631198002a <unknown>\n#1 0x5563116665e0 <unknown>\n#2 0x55631165e7eb <unknown>\n#3 0x55631164ea09 <unknown>\n#4 0x556311650744 <unknown>\n#5 0x55631164ecbd <unknown>\n#6 0x55631164e52f <unknown>\n#7 0x55631164e422 <unknown>\n#8 0x55631164c493 <unknown>\n#9 0x55631164cada <unknown>\n#10 0x556311668f41 <unknown>\n#11 0x5563116fa8c5 <unknown>\n#12 0x5563116dab22 <unknown>\n#13 0x5563116f9d7d <unknown>\n#14 0x5563116da8c3 <unknown>\n#15 0x5563116a86b3 <unknown>\n#16 0x5563116a968e <unknown>\n#17 0x55631194aa2b <unknown>\n#18 0x55631194e9b1 <unknown>\n#19 0x556311937225 <unknown>\n#20 0x55631194f532 <unknown>\n#21 0x55631191c38f <unknown>\n#22 0x55631196ef28 <unknown>\n#23 0x55631196f0f3 <unknown>\n#24 0x55631197ee7c <unknown>\n#25 0x7f0e56894ac3 <unknown>\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/auto-tests-python/auto-tests-python/tests/conftest.py\", line 22, in main_page_open\n    driver.get(MainPageLinks.URL_MAIN_PAGE)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 353, in get\n    self.execute(Command.GET, {\"url\": url})\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 344, in execute\n    self.error_handler.check_response(response)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "driver",
        "value" : "<selenium.webdriver.chrome.webdriver.WebDriver (session=\"2fadf8c0f6d698aa078854f54815132a\")>"
      } ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 1,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  } ],
  "afterStages" : [ {
    "name" : "driver::0",
    "time" : {
      "start" : 1728610986580,
      "stop" : 1728610986642,
      "duration" : 62
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Main Page."
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "main_page_test"
  }, {
    "name" : "subSuite",
    "value" : "TestMainPage"
  }, {
    "name" : "host",
    "value" : "fv-az740-570"
  }, {
    "name" : "thread",
    "value" : "2375-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.main_page_test"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 2,
        "skipped" : 0,
        "passed" : 57,
        "unknown" : 0,
        "total" : 59
      },
      "items" : [ {
        "uid" : "bd5d17b1d970664",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/58//#testresult/bd5d17b1d970664",
        "status" : "passed",
        "time" : {
          "start" : 1728524864261,
          "stop" : 1728524864915,
          "duration" : 654
        }
      }, {
        "uid" : "893c0a3be7b5fd1",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/57//#testresult/893c0a3be7b5fd1",
        "status" : "passed",
        "time" : {
          "start" : 1728438538465,
          "stop" : 1728438539114,
          "duration" : 649
        }
      }, {
        "uid" : "5af29ad58b2d6173",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/56//#testresult/5af29ad58b2d6173",
        "status" : "passed",
        "time" : {
          "start" : 1728352114426,
          "stop" : 1728352115074,
          "duration" : 648
        }
      }, {
        "uid" : "eed5e6b32a94a361",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/55//#testresult/eed5e6b32a94a361",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/specialists",
        "time" : {
          "start" : 1728266372902,
          "stop" : 1728266413305,
          "duration" : 40403
        }
      }, {
        "uid" : "f15d61231932e6f9",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/54//#testresult/f15d61231932e6f9",
        "status" : "passed",
        "time" : {
          "start" : 1728179659085,
          "stop" : 1728179659746,
          "duration" : 661
        }
      }, {
        "uid" : "1179aa12fc25baf7",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/53//#testresult/1179aa12fc25baf7",
        "status" : "passed",
        "time" : {
          "start" : 1728092880664,
          "stop" : 1728092881308,
          "duration" : 644
        }
      }, {
        "uid" : "77fcaff26120dd57",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/52//#testresult/77fcaff26120dd57",
        "status" : "passed",
        "time" : {
          "start" : 1728006583103,
          "stop" : 1728006583747,
          "duration" : 644
        }
      }, {
        "uid" : "aa22c53ff225eca4",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/51//#testresult/aa22c53ff225eca4",
        "status" : "passed",
        "time" : {
          "start" : 1727920216155,
          "stop" : 1727920216839,
          "duration" : 684
        }
      }, {
        "uid" : "aa7e523b4516f2f5",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/50//#testresult/aa7e523b4516f2f5",
        "status" : "passed",
        "time" : {
          "start" : 1727833670736,
          "stop" : 1727833671403,
          "duration" : 667
        }
      }, {
        "uid" : "fd19cb8001157bdb",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/49//#testresult/fd19cb8001157bdb",
        "status" : "passed",
        "time" : {
          "start" : 1727747794654,
          "stop" : 1727747795342,
          "duration" : 688
        }
      }, {
        "uid" : "73b677110428387c",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/48//#testresult/73b677110428387c",
        "status" : "passed",
        "time" : {
          "start" : 1727661078184,
          "stop" : 1727661078833,
          "duration" : 649
        }
      }, {
        "uid" : "a21063c8b4fc9722",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/47//#testresult/a21063c8b4fc9722",
        "status" : "passed",
        "time" : {
          "start" : 1727574899973,
          "stop" : 1727574900627,
          "duration" : 654
        }
      }, {
        "uid" : "93455f58c304af32",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/46//#testresult/93455f58c304af32",
        "status" : "passed",
        "time" : {
          "start" : 1727488127641,
          "stop" : 1727488128313,
          "duration" : 672
        }
      }, {
        "uid" : "304d19e3ce2f2ab6",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/45//#testresult/304d19e3ce2f2ab6",
        "status" : "passed",
        "time" : {
          "start" : 1727401834996,
          "stop" : 1727401835645,
          "duration" : 649
        }
      }, {
        "uid" : "41cd524348dcf23c",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/44//#testresult/41cd524348dcf23c",
        "status" : "passed",
        "time" : {
          "start" : 1727315319767,
          "stop" : 1727315320411,
          "duration" : 644
        }
      }, {
        "uid" : "108661c69ecadca1",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/43//#testresult/108661c69ecadca1",
        "status" : "passed",
        "time" : {
          "start" : 1727228970260,
          "stop" : 1727228970909,
          "duration" : 649
        }
      }, {
        "uid" : "297267207af2190f",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/42//#testresult/297267207af2190f",
        "status" : "passed",
        "time" : {
          "start" : 1727142518506,
          "stop" : 1727142519155,
          "duration" : 649
        }
      }, {
        "uid" : "32f237c725f5798b",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/41//#testresult/32f237c725f5798b",
        "status" : "passed",
        "time" : {
          "start" : 1727056048140,
          "stop" : 1727056048807,
          "duration" : 667
        }
      }, {
        "uid" : "e2def013ec7bd048",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/40//#testresult/e2def013ec7bd048",
        "status" : "passed",
        "time" : {
          "start" : 1726969977108,
          "stop" : 1726969977758,
          "duration" : 650
        }
      }, {
        "uid" : "d3d4f7908a7528e8",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/39//#testresult/d3d4f7908a7528e8",
        "status" : "passed",
        "time" : {
          "start" : 1726882978040,
          "stop" : 1726882978704,
          "duration" : 664
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "c35103994b0c3821.json",
  "parameterValues" : [ ]
}